### PR TITLE
Removed Halogen specific files and combined functionality where possible

### LINKED ISSUE #15: Investigate Halogen specific modules FindCoreFormulae2_Halogen and Utils_Halogen
Logic related to halogen compounds should be investigated to see which atoms are support and what kind of functionality is implemented to see what needs to be extended.

### PR DESCRIPTION
**Pull request by hechth**
This PR removes the Halogen specific files and tries to combine the duplicated functions where possible.
I also implemented a test for the MFAssign function since this one actually uses those functions which I combined or deleted in case they were identical to the base functions.

We have to decide whether we even need the wrapper for the purely organic molecules when thinking about Galaxy later on.

Ideally, the MFAssign function would re-use the code for MFAssignCHO, which I think it does not do at this stage, and then the code for various elements would call different functions etc.

Closes #15

### CODE REVIEW

File: MFAssignR/DESCRIPTION [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+Depends: R (>= 3.4.1)
+Imports: dplyr (>= 0.7.6),
+         tidyr (>= 0.8.1),
+         ggplot2 (>= 3.0.0),
+	 	 colorRamps (>= 2.3)
+RoxygenNote: 7.2.3
+Suggests: knitr,
+    	  rmarkdown,
+		  datacomparer (>= 0.1.4),
+		  patrick (>= 0.2.0)
```
Comment (wverastegui) [inline MFAssignR/DESCRIPTION:29]:
```suggestion
         ggplot2 (>= 3.0.0),
	 colorRamps (>= 2.3)
RoxygenNote: 7.2.3
Suggests: knitr,
    	  rmarkdown,
	  datacomparer (>= 0.1.4),
	  patrick (>= 0.2.0)
```

File: MFAssignR/R/FindCoreFormulae2.R [hunk 2]
```diff
@@ -12,74 +12,110 @@
...
-#maxAttempts  <- 10
-FindCoreFormulae2 <- function(env) {
-  # constants
-  #coreXEM <- 102.0322
-  #coreRNM <- 102
-  #RECORDS <- vector("list", 1000000)
-  moietyMass <- 0.0363855087200022;
-  maxAttempts <- env$Ox/3;
+#' @export
+FindCoreFormulae2_Halo <- function(env) {
```
Comment (zargham-ahmad) [inline MFAssignR/R/FindCoreFormulae2.R:54]: FindCoreFormulae2 and FindCoreFormulae2_Halo looks quite similar

File: MFAssignR/tests/testthat/test-MFAssign.R [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+patrick::with_parameters_test_that("MFAssign works",
+  {
+    peaks <- readRDS(file.path("test-data", paste0(mode, "_iso.rds")))
+    actual <- MFAssign(
+      peaks = peaks$Mono,
+      isopeaks = peaks$Iso,
+      ionMode = mode
+    )
+    save_expected(actual, mode)
```
Comment (zargham-ahmad) [inline MFAssignR/tests/testthat/test-MFAssign.R:33]: Are we saving the expected file on each run?

**Review by zargham-ahmad — COMMENTED** [review]
Look good to me!

File: MFAssignR/R/FindCoreFormulae2.R [hunk 4]
```diff
@@ -12,74 +12,110 @@
...
+#' Assigns all possible combinations of C, H, and O for a given MW
+#'
+#' FindCoreFormulae assigns all possible combinations of C, H, and O for a
+#' given MW and subject to constraints in the \code{\link{ValidFormula}} function.
+#' This is a subfunction of the MFAssign function and will not work
+#' independently of that function.
 #'
+#' @param env environment from \code{\link{MFAssign}} function
 #'
+#' @examples
```
Comment (KristinaGomoryova) [inline MFAssignR/R/FindCoreFormulae2.R:49]: examples can be deleted, they caused issues in devtools::check()

**Review by KristinaGomoryova — COMMENTED** [review]
I would just delete the examples as they were an issue when running a check on the package, otherwise it looks good!

File: MFAssignR/tests/testthat/test-MFAssign.R (hunk 3 above)
Comment (hechth) [reply MFAssignR/tests/testthat/test-MFAssign.R:33]: Good catch!

**Pull request merged**